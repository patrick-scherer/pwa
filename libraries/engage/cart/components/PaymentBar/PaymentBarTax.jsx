// @flow
import * as React from 'react';
import { SurroundPortals } from '@shopgate/engage/components';
import {
  CART_PAYMENT_BAR_TOTALS_TAX,
  getTaxLine,
} from '@shopgate/pwa-common-commerce/cart';
import CartTotalLine from '@shopgate/pwa-ui-shared/CartTotalLine';
import { CartContext } from '../../cart.context';
import { spacer } from './PaymentBarContent.style';
import connect from './PaymentBarTax.connector';

type TaxData = {
  label: string,
  amount: number,
}

type Props = {
  taxData?: TaxData,
  showSeparator?: boolean,
  className?: string,
}

/**
 * The Tax component.
 * @returns {JSX}
 */
function PaymentBarTax({ taxData, showSeparator, className }: Props) {
  const {
    currency, isLoading, config, hasPromotionCoupons,
  } = React.useContext(CartContext);

  if (!taxData) {
    return null;
  }

  const taxLine = getTaxLine(config, taxData);

  if (!taxLine) {
    return null;
  }

  return (
    <SurroundPortals portalName={CART_PAYMENT_BAR_TOTALS_TAX}>
      <CartTotalLine isDisabled={isLoading} type="tax" className={className}>
        <CartTotalLine.Label label={taxLine.label} showSeparator={showSeparator} />
        <CartTotalLine.Amount amount={taxLine.amount} currency={currency} />
        <CartTotalLine.Hint hint={taxLine.hint} />
        { hasPromotionCoupons && (
          <CartTotalLine.Spacer className={spacer} />
        )}
      </CartTotalLine>
    </SurroundPortals>
  );
}

PaymentBarTax.defaultProps = {
  taxData: null,
  className: null,
  showSeparator: true,
};

export default connect(PaymentBarTax);
