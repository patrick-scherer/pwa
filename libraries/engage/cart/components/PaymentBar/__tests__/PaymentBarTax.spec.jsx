import React from 'react';
import { shallow } from 'enzyme';
import PaymentBarTax from '../PaymentBarTax';

jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useContext: () => ({
    currency: 'EUR',
    config: {
      hideTax: false,
      tax: {
        text: null,
        hint: null,
      },
    },
  }),
}));

jest.mock('../PaymentBarTax.connector', () => cmp => cmp);

jest.mock('@shopgate/engage/components');

describe('<PaymentBarTax />', () => {
  it('should render and match snapshot', () => {
    // eslint-disable-next-line extra-rules/no-single-line-objects
    const wrapper = shallow(<PaymentBarTax taxData={{ label: 'Tax', amount: 10 }} />).dive();
    expect(wrapper).toMatchSnapshot();
  });
  it('should not render and match snapshot', () => {
    const wrapper = shallow(<PaymentBarTax taxData={null} />);
    expect(wrapper).toBeEmptyRender();
  });
});
