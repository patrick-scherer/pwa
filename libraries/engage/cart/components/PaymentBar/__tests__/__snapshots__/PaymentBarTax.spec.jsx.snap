// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PaymentBarTax /> should render and match snapshot 1`] = `
<CartTotalLine
  className={null}
  isDisabled={false}
  type="tax"
>
  <Label
    label="Tax"
    labelParams={Object {}}
    showSeparator={true}
  />
  <Amount
    amount={10}
    currency="EUR"
  />
  <Hint
    hint={null}
  />
</CartTotalLine>
`;
